client/src/components/ElementActions/tests/PublishAction-test.js

```diff
@@ -1,109 +1,93 @@
-  const ActionComponent = PublishAction(WrappedComponent);
-  const jQuery = jest.fn();
-  window.jQuery = jQuery;
-
-  beforeEach(() => {
-    mockMutation = jest.fn(() => new Promise((resolve) => resolve()));
-    wrapper = mount(
-      <ActionComponent
-        title="My abstract action"
-        element={{
-          id: 123,
-          version: 234,
-          liveVersion: false,
-          blockSchema: { type: 'Test' }
-        }}
-        type={{ broken: false }}
-        actions={{ handlePublishBlock: mockMutation }}
-        toggle={false}
-      />
-    );
-
-    jQuery.noticeAdd = jest.fn();
-  });
-
-  it('renders the wrapped component', () => {
-    expect(wrapper.children().first().type()).toEqual(WrappedComponent);
-  });
-
-  it('renders a button', () => {
-    expect(wrapper.find('button').length).toBe(1);
-  });
-
-  it('renders the title and class', () => {
-    expect(wrapper.find('button').text()).toContain('Publish');
-    expect(wrapper.find('button').hasClass('element-editor__actions-publish')).toBe(true);
-  });
-
-  it('publishes from draft to live', async () => {
-    wrapper.find('button').simulate('click');
-
-    // The click handler does not return a promise, but it does USE one.
-    // We need to await the promise resolution cycle before asserting.
-    await new Promise((resolve) => resolve());
-
-    expect(mockMutation).toHaveBeenCalledWith(123);
-  });
-
-  it('returns null when is the live version', () => {
-    const draftWrapper = mount(
-      <ActionComponent
-        element={{ isLiveVersion: true, blockSchema: { type: 'Test' } }}
-        type={{ broken: false }}
-        actions={{ handlePublishBlock: mockMutation }}
-      />
-    );
+window.jQuery = {
+  noticeAdd: () => null
+};
+
+const WrappedComponent = (props) => <div>{props.children}</div>;
+const ActionComponent = PublishAction(WrappedComponent);
+
+function makeProps(obj = {}) {
+  return {
+    element: {
+      id: 123,
+      version: 234,
+      liveVersion: false,
+      blockSchema: { type: 'Test' }
+    },
+    type: {
+      broken: false
+    },
+    actions: {
+      handlePublishBlock: () => {}
+    },
+    toggle: false,
+    ...obj,
+  };
+}
+
+test('PublishAction renders the wrapped component', () => {
+  const { container } = render(<ActionComponent {...makeProps()}/>);
+  expect(container.querySelector('button.element-editor__actions-publish').textContent).toBe('Publish');
+});
 
-    expect(draftWrapper.find('button').length).toBe(0);
-  });
+test('PublishAction publishes from draft to live', async () => {
+  const mockMutation = jest.fn(() => new Promise((resolve) => resolve()));
+  const { container } = render(
+    <ActionComponent {...makeProps({
+      actions: {
+        handlePublishBlock: mockMutation
+      }
+    })}
+    />
+  );
+  fireEvent.click(container.querySelector('button.element-editor__actions-publish'));
+  await new Promise((resolve) => resolve());
```

Comment:
The test will fail if it's removed, that will prevent people from removing it :-)

This is a type of thing I've done loads of times. Best guess as to why I did it is because RTL complained that something wasn't "wrapped in act()" because there was some component state update than happened then a render, or something like that. 

I could put in a note as to why this is done, but it's done it loads of jest tests therefore either I document why it was done like this everywhere, or nowhere.

I think the simple fact that you can delete it, run the test and see the bizarre failure is enough explanation of why we need this code. I myself cannot really articulate the why because thing is really convulated

Reply:
I don't like that it's a bizarre failure, a comment would save future devs some time debugging that - but I take your point that this is in multiple places, so a comment here won't be so useful. Given that context, this is fine as is.